### PR TITLE
python3Packages.libpurecool: init at 0.6.4

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python library for Dyson devices.

https://github.com/etheralm/libpurecool

This is a Home Assistant dependency.

I removed the vendorized zeroconf. Thus, it would be nice if one with access to the actual hardware could test it. Thanks.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112403` at 37bcfbc6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/22ca884f780cb7d609d35fea2b3aa706">python37Packages.libpurecool</a></li></li>
    <li><a href="https://gist.github.com/fec2a25fade5db3508e0935ee6c91d56">python38Packages.libpurecool</a></li></li>
    <li><a href="https://gist.github.com/b68d551213193fb928809deec4ce50ec">python39Packages.libpurecool</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112403` at 37bcfbc6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/7308faf3a9440f4c05df01d1f2155009">python37Packages.libpurecool</a></li></li>
    <li><a href="https://gist.github.com/0317a8f0251eac838558ec5e21bd0680">python38Packages.libpurecool</a></li></li>
    <li><a href="https://gist.github.com/44039bc6c21791d5e994e54c9346f10e">python39Packages.libpurecool</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/libpurecool/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  version = "0.6.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1kwbinbg0i4fca1bpx6jwa1fiw71vg0xa89jhq4pmnl5cn9c8kqx";
+  };
+
+  # Remove vendorized zeroconf, https://github.com/etheralm/libpurecool/issues/33
+  postPatch = ''
+    rm -rf libpurecool/zeroconf.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/libpurecool/default.nix:23]:
```suggestion
    rm libpurecool/zeroconf.py
```

File: pkgs/development/python-modules/libpurecool/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    substituteInPlace libpurecool/dyson_pure_cool_link.py --replace \
+      "from .zeroconf import ServiceBrowser, Zeroconf" "from zeroconf import ServiceBrowser, Zeroconf"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/libpurecool/default.nix:25]:
```suggestion
    substituteInPlace libpurecool/dyson_pure_cool_link.py \
      --replace "from .zeroconf import ServiceBrowser, Zeroconf" "from zeroconf import ServiceBrowser, Zeroconf"
```

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -3660,6 +3660,8 @@ in {
     inherit python;
   })).py;
 
+  libpurecool= callPackage ../development/python-modules/libpurecool { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3663]:
```suggestion
  libpurecool = callPackage ../development/python-modules/libpurecool { };
```

File: pkgs/development/python-modules/libpurecool/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/libpurecool/default.nix:23]: I remember...we want it to fail if the file is removed with the next release.

Thanks for pointing it out again.

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/top-level/python-packages.nix]: Fixed, thanks.

File: pkgs/development/python-modules/libpurecool/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/libpurecool/default.nix]: Changed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112403` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.libpurecool</li>
<li>python38Packages.libpurecool</li>
<li>python39Packages.libpurecool</li>
</ul>
</details>

**Pull request merged**